### PR TITLE
OpenID4VCI: Validate grants in credential offer

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes https://github.com/nuts-foundation/nuts-node/issues/2038

### CODE REVIEW

File: vcr/holder/oidc4vci_wallet.go [hunk 1]
```diff
@@ -146,6 +155,23 @@ func (h wallet) HandleCredentialOffer(ctx context.Context, offer oidc4vci.Creden
 	return nil
 }
 
+func getPreAuthorizedCodeFromOffer(offer oidc4vci.CredentialOffer) string {
```
Comment (beardedfoo) [inline vcr/holder/oidc4vci_wallet.go:158]: This was a nice function to make 👍

File: vcr/holder/oidc4vci_wallet_test.go [hunk 2]
```diff
@@ -126,6 +129,40 @@ func Test_wallet_HandleCredentialOffer(t *testing.T) {
...
+	t.Run("pre-authorized code grant", func(t *testing.T) {
+		w := NewOIDCWallet(holderDID, "https://holder.example.com", nil, nil, nil, time.Second*5).(*wallet)
+		t.Run("no grants", func(t *testing.T) {
+			offer := oidc4vci.CredentialOffer{Credentials: []map[string]interface{}{{}}}
+			err := w.HandleCredentialOffer(audit.TestContext(), offer)
+			require.EqualError(t, err, "invalid_grant - couldn't find (valid) pre-authorized code grant in credential offer")
+		})
+		t.Run("no pre-authorized grant", func(t *testing.T) {
+			offer := oidc4vci.CredentialOffer{
+				Credentials: []map[string]interface{}{{}},
```
Comment (beardedfoo) [inline vcr/holder/oidc4vci_wallet_test.go:141]: Is there any way to simplify this? Maybe defining a type or two above? This looks like something from a golang proficiency exam

File: vcr/holder/oidc4vci_wallet.go [hunk 3]
```diff
@@ -102,7 +111,7 @@ func (h wallet) HandleCredentialOffer(ctx context.Context, offer oidc4vci.Creden
 	accessTokenResponse, err := issuerClient.RequestAccessToken(oidc4vci.PreAuthorizedCodeGrant, map[string]string{
 		// TODO: The code below is unsafe, validate offered grants and then extract the pre-authorized code
```
Comment (woutslakhorst) [inline vcr/holder/oidc4vci_wallet.go:112]: remove comment

File: vcr/holder/oidc4vci_wallet_test.go (hunk 2 above)
Comment (reinkrul) [reply vcr/holder/oidc4vci_wallet_test.go:141]: Wrapped it in a function with docs.

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**